### PR TITLE
[SKIP CI] doxygen fixes

### PR DESCRIPTION
**Pull request by marc-hb**

### CODE REVIEW

**Review by cujomalainey — APPROVED** [review]
LGTM for crossover, any chance we can have these kind of issues error out in CI?

**Comment by marc-hb** [comment]
> LGTM for crossover, any chance we can have these kind of issues error out in CI?

That would require:
1. Automating a second doxygen configuration similar to the one I'm using to extract everything (doxygen is a very good code indexer, call graphs especially rock). Not rocket science, just a matter of finding some time to do it.
2. Much more problematically, that would require fixing or reliably excluding all the other errors left. As I mentioned in one of the commit message, right now I'm using some ugly and unreliable `grep -v` that loses the exit status and barely does the job, leaving a (reasonable) number of errors make it through. `WARN_IF_INCOMPLETE_DOC` would go a long way but it won't be available before doxygen 1.10.

tl;dr: it should be possible but a significant amount of work.

Just for the record here's a raw dump of my current config:
```diff
+GENERATE_XML    = NO
+
+# CASE_SENSE_NAMES = NO
+INPUT            = @top_srcdir@/src/
 EXCLUDE                 = @top_srcdir@/src/include/sof/audio/MaxxEffect
 RECURSIVE       = YES
 FILE_PATTERNS    = *.c *.h
@@ -18,13 +14,21 @@ EXAMPLE_PATH     = @top_srcdir@/test
 IMAGE_PATH      =
 QUIET            = YES
 
+DOT_IMAGE_FORMAT = svg
+INTERACTIVE_SVG = YES
+
 EXTRACT_ALL     = YES
 EXTRACT_STATIC  = YES
+
 WARN_IF_UNDOCUMENTED = NO
-SHOW_INCLUDE_FILES = NO
+# Needs doxygen 1.10
+WARN_IF_INCOMPLETE_DOC = NO
+WARN_NO_PARAMDOC = NO
+
+SHOW_INCLUDE_FILES = YES
 JAVADOC_AUTOBRIEF = YES
 INHERIT_DOCS    = YES
-ENABLED_SECTIONS = ""
+# ENABLED_SECTIONS = ""
 MACRO_EXPANSION  = YES
 EXPAND_ONLY_PREDEF = YES
 PREDEFINED      = "__attribute__(x)= " \
@@ -32,10 +36,18 @@ PREDEFINED   = "__attribute__(x)= " \
                   "CONFIG_FORMAT_S32LE=1" \
                   "CONFIG_TRACE=1"
 
+SOURCE_BROWSER = YES
+STRIP_CODE_COMMENTS = NO
+
+HAVE_DOT = YES
+
+CALL_GRAPH = YES
+CALLER_GRAPH = YES
+
 OPTIMIZE_OUTPUT_FOR_C = YES
 TYPEDEF_HIDES_STRUCT = YES
 
 #INPUT_FILTER   = inputfilter
 #FILTER_SOURCE_FILES = YES
 
-HTML_TIMESTAMP = NO
+# HTML_TIMESTAMP = NO

```

**Review by lgirdwood — APPROVED** [review]
